### DISCUSSION TITLE
routing not working near saudi arabia to qatar

### DISCUSSION THREAD

**Discussion by mkandulavm** [valhalla/valhalla#4868] [Q&A]
Hi

Found a strange issue

this route works
https://www.openstreetmap.org/directions?engine=fossgis_valhalla_car&route=25.4287%2C50.8511%3B25.2584%2C49.8965#map=10/25.0788/50.4396

and this fails
https://www.openstreetmap.org/directions?engine=fossgis_valhalla_car&route=25.4287%2C50.8511%3B25.2725%2C49.8568

For the failing route, if we try with graphhopper , it shows the route.

I think this requires some investigation and a test case as a followup ?

**Comment by mkandulavm** [comment]
Hi. Can I get some feedback on this please ?

**Comment by chrstnbwnkl** [comment]
Seems like the root cause is that hierarchy changes are necessary here at the border crossing: https://www.openstreetmap.org/search?lat=24.71845&lon=50.75847#map=16/24.71846/50.75847, plus the long detour to get to get to the other side of the road in combination lead to the reverse search branch maxing out and not finding a route. That's what the expansion suggests: 

```
{
	"locations": [
		{
			"lat": 25.4287,
			"lon": 50.8511
		},
		{
			"lat": 25.2725,
			"lon": 49.8568
		}
	],
	"costing": "auto",
	"directions_options": {
		"units": "km",
		"language": "en"
	},
	"action": "route"
}
```


I wonder if Valhalla could improve this by reclassifying these border control area streets as motorways? Either that or it's a mapping error, I know that at other border crossings these roads are mapped as motorways: https://www.openstreetmap.org/#map=19/32.542120/-117.028695

**Reply by chrstnbwnkl** [reply]
Converted this to a discussion for now, I don't see how this is necessary an issue with Valhalla (yet)

**Comment by ObaidDev** [comment]
thank @chrstnbwnkl  for you support .
is there any way to get the optimized route if the distance is greater than 200 km?

**Reply by chrstnbwnkl** [reply]
Seems unrelated to this thread :smile: optimized route uses `CostMatrix` under the hood, so you need to fiddle with the matrix service limits for the profile(s) you want to use
